Cesium3DTiles/src/RasterizedPolygonsOverlay.cpp

```diff
@@ -0,0 +1,214 @@
+              (v_proj_ab_perp <= 0.0 && v_proj_ca_perp <= 0.0 &&
+               v_proj_bc_perp <= 0.0)) {
+            image.pixelData[256 * j + i] = static_cast<std::byte>(0xff);
+          }
+        }
+      }
+    }
+  }
+}
+} // namespace
+
+class CESIUM3DTILES_API RasterizedPolygonsTileProvider final
+    : public RasterOverlayTileProvider {
+
+private:
+  std::vector<CartographicSelection> _polygons;
+  std::string _textureTargetName;
+  uint32_t _currentTileID;
+
+public:
+  RasterizedPolygonsTileProvider(
+      RasterOverlay& owner,
+      const CesiumAsync::AsyncSystem& asyncSystem,
+      const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
+      const std::shared_ptr<IPrepareRendererResources>&
+          pPrepareRendererResources,
+      const std::shared_ptr<spdlog::logger>& pLogger,
+      const CesiumGeospatial::Projection& projection,
+      const std::string& textureTargetName,
+      const std::vector<CartographicSelection>& polygons)
+      : RasterOverlayTileProvider(
+            owner,
+            asyncSystem,
+            pAssetAccessor,
+            std::nullopt,
+            pPrepareRendererResources,
+            pLogger,
+            projection),
+        _textureTargetName(textureTargetName),
+        _polygons(polygons),
+        _currentTileID(0) {}
+
+  virtual void mapRasterTilesToGeometryTile(
+      const CesiumGeospatial::GlobeRectangle& geometryRectangle,
+      double targetGeometricError,
+      std::vector<std::unique_ptr<Cesium3DTiles::RasterMappedTo3DTile>>&
+          outputRasterTiles,
+      std::optional<size_t> outputIndex) override {
+    this->mapRasterTilesToGeometryTile(
+        projectRectangleSimple(this->getProjection(), geometryRectangle),
+        targetGeometricError,
+        outputRasterTiles,
+        outputIndex);
+  }
+
+  virtual void mapRasterTilesToGeometryTile(
+      const CesiumGeometry::Rectangle& geometryRectangle,
+      double /*targetGeometricError*/,
+      std::vector<std::unique_ptr<Cesium3DTiles::RasterMappedTo3DTile>>&
+          outputRasterTiles,
+      std::optional<size_t> /*outputIndex*/) override {
+    if (this->_pPlaceholder) {
+      outputRasterTiles.push_back(
+          std::make_unique<QuadtreeRasterMappedTo3DTile>(
+              this->_pPlaceholder.get(),
+              CesiumGeometry::Rectangle(0.0, 0.0, 0.0, 0.0)));
+      return;
+    }
+
+    uint32_t tileID = this->_currentTileID++;
+
+    outputRasterTiles.push_back(std::make_unique<QuadtreeRasterMappedTo3DTile>(
+        CesiumUtility::IntrusivePointer<RasterOverlayTile>(
+            this->getTile(tileID, geometryRectangle)),
+        geometryRectangle));
+  }
+
+  virtual bool
+  hasMoreDetailsAvailable(const TileID& /*tileID*/) const override {
+    return false; // always true or always false??
+  }
+
+  virtual CesiumAsync::Future<LoadedRasterOverlayImage>
+  loadTileImage(const TileID& tileID) override {
+    CesiumUtility::IntrusivePointer<RasterOverlayTile> pTile =
```

Comment:
`loadTileImage` will be called from the main/game thread. So you probably should use `thenInWorkerThread` to rasterize the polygon in a worker.